core/arch/aarch64/codec.py

```diff
@@ -98,39 +116,63 @@ def gen(c, pats, depth):
             if x < best_x:
                 best_b = b
                 best_x = x
-        c.append('%sif ((enc >> %d & 1) == 0) {' % (indent, best_b))
+        c += ['%sif ((enc >> %d & 1) == 0) {' % (indent, best_b)]
         pats0 = []
         pats1 = []
         for p in pats:
             (f, v, _, _) = p
             if (1 << best_b) & (~f | v):
-                pats0.append(p)
+                pats0 += [p]
             if (1 << best_b) & (f | v):
-                pats1.append(p)
+                pats1 += [p]
         gen(c, pats0, depth + 1)
-        c.append('%s} else {' % indent)
+        c += ['%s} else {' % indent]
         gen(c, pats1, depth + 1)
-        c.append('%s}' % indent)
+        c += ['%s}' % indent]
```

Comment:
All of these s/append/+=/ seem to confuse the issue here: unrelated to the content changes, right?  Maybe better to be a separate CL.